src/main/java/programminglife/model/Node.java

```diff
@@ -37,13 +40,13 @@ public void setSequence(String sequence) {
         this.sequence = sequence;
     }
 
-    public static Node parseSegment(Scanner sc) {
-        int id = Integer.parseInt(sc.next());
-        String segment = sc.next();
-        sc.next(); // Skip unused '*' symbol
-        String origin = sc.next();
-        String offset = sc.next();
-        String readCount = sc.next();
+    public static Node parseSegment(String propertyString) {
+        String[] properties = propertyString.split("\\s");
+        // properties[0] is 'S'
```

Comment:
Same as above for `parseLink`
Instead if simply assuming, either add an assert (can we get Travis to run with assertions enabled?) or an explicit if
Add this as a precondition for the method in the JavaDoc either way.